### DISCUSSION TITLE
How to create SharedValue without useSharedValue hook?

### DISCUSSION THREAD

**Discussion by iqqmuT** [software-mansion/react-native-reanimated#1901] [Q&A]
I have a case where I want to update SharedValues from Redux Saga. I don't want to re-render the React components because it is a heavy operation. I want to update their UI attributes (e.g. position, Z index) directly from Redux Saga code.

Is there a way to create SharedValues without `useSharedValue` hook? I would like to create SharedValues in Saga code and let the React components use them.

**Comment by pkvov** [comment]
i have the same problem here.
it's painful to create bunch of SharedValue through useSharedValue.
[Rules of Hooks](https://reactjs.org/docs/hooks-rules.html)
**Don’t call Hooks inside loops, conditions, or nested functions. Instead, always use Hooks at the top level of your React function, before any early returns.** 

when i create SharedValue inside loops, then get the React error
Error: Rendered fewer hooks than expected. This may be caused by an accidental early return statement.

Is there a way to create SharedValues without useSharedValue hook?

**Comment by anhkieet** [comment]
Same here. I want to create sharedvalue foreach item in array. How can I do this without using hook? @kmagiera

**Comment by aldovandus** [comment]
I have the same problem. I found a solution but i don't know if it is the best:

`import { makeMutable } from 'react-native-reanimated'`

`const xPosition = makeMutable(0)`

`xPosition.value = 10`

I used it with redux and it works.
